src/utils.jl

```diff
@@ -0,0 +1,46 @@
+
+"""
+    ignore() do
+        ...
+    end
+
+Tell ChainRules to ignore a block of code. Everything inside the `do` block will run
+on the forward pass as normal, but ChainRules won't try to differentiate it at all.
+This can be useful for e.g. code that does logging of the forward pass.
+
+Obviously, you run the risk of incorrect gradients if you use this incorrectly.
+"""
+ignore(f) = f()
+
+"""
+    @ignore (...)
+
+Tell ChainRules to ignore an expression. Equivalent to `ignore() do (...) end`.
+Example:
+
+```julia-repl
+julia> f(x) = x
+julia> _, v_pullback = ChainRules.rrule(ChainRules.ignore, f, (1,)...)
+julia> _, v̄ = v_pullback(1)
+julia> v̄
+nothing
+```
+"""
+
+function frule((_, ḟ), ::typeof(ignore), f, args...; kwargs...)
+    return f(args...; kwargs...), nothing
+end
+
+function rrule(::typeof(ignore), f, args...; kwargs...)
+    y = f(args...; kwargs...)
+    function ignore_pullback(ȳ)
+        return (NO_FIELDS, nothing)
```

Comment:
Same here. Additionally, we need one `Zero` per-argument. @oxinabox I always forget what the most efficient way to generate a collection of `Zero`s of the correct size is likely to be -- could you adivse?

Reply:
Would that be simply:
```jl
julia> zeros(2, 2)
2×2 Array{Float64,2}:
 0.0  0.0
 0.0  0.0
```
i.e. `zeros(length(args), length(args))`?

Reply:
Ah, wait, no. I misspoke.

Since the only method of `ignore` is unary, this `rrule` should also be unary, and doesn't need to have any `kwargs`. For example, you could simply write it as
```julia
function rrule(::typeof(ignore), f)
    function ignore_pullback(ȳ)
        return (NO_FIELDS, Zero())
    end
    return f(), ignore_pullback
end
```

Note the use of `Zero()` as the sensitivity of `f`, rather than `nothing`.